httpclient5/src/main/java/org/apache/hc/client5/http/impl/io/LenientHttpResponseParser.java

```diff
@@ -85,7 +85,7 @@ protected ClassicHttpResponse createMessage(final CharArrayBuffer buffer) throws
             return super.createMessage(buffer);
         } catch (final HttpException ex) {
             if (this.log.isDebugEnabled()) {
-                this.log.debug("Garbage in response: " + buffer.toString());
+                this.log.debug("Garbage in response: {}", buffer.toString());
```

Comment:
same here

Reply:
@carterkozak Generally the overhead of string concatenation should not matter that much given all log entries are guarded with `#is<LogLevel>Enabled` but I am fine with using more native `slf4j` notation.

@michael-o Shall I merge the PR or is there still something that needs work?

Reply:
@ok2c It requires some adjustments by @carterkozak.

Reply:
> Generally the overhead of string concatenation should not matter that much given all log entries are guarded with #is<LogLevel>Enabled but I am fine with using more native slf4j notation.

In cases where debug logging is enabled we can write logs to disk without any allocations using log4j2 :-)
See https://logging.apache.org/log4j/2.x/manual/garbagefree.html

> It requires some adjustments by @carterkozak.

I've applied the intellij automation to avoid unnecessary toString calls.

Reply:
Separately (with a separate ticket), I'd like to consider replacing instance loggers with static loggers to avoid paying a penalty on instantiation. This would result in slightly different (and arguably better) behavior when classes are extended, as the logger origin would reflect the location that produced logging, not the subclass that happened to extend it. For example, if I want to debug hc5, I might set the `org.apache.hc` origin to `TRACE` level and add it to a separate appender. If I've extended a component it won't appear with my httpclient logs using the `getClass()` approach.
Thoughts on this proposal?

Reply:
@carterkozak Lets complete this PR first, I am open to new ones.

Reply:
@michael-o Agreed, I think I've resolved all the requests. Note I did not make changes to header iteration since it's sufficiently outside the scope of this PR, and if we're confident the list implements RandomAccess would produce slightly worse performance due to iterator allocation.

Reply:
Let me brisky go through again. Iteration is defitively a new PR. For your other proposal just create new JIRA issue please.

Reply:
> Separately (with a separate ticket), I'd like to consider replacing instance loggers with static loggers to avoid paying a penalty on instantiation. ...
> Thoughts on this proposal?

@carterkozak As far as I know static loggers in low level libraries are considered evil and are to be avoided in favor of instance loggers. The overhead of logger instantiation is quite manageable if one injects logger through a constructor of frequently allocated and deallocated components.

Reply:
I've created this ticket to centralize discussion: https://issues.apache.org/jira/browse/HTTPCLIENT-2096